src/transformers/pipelines/image_to_text.py

```diff
@@ -98,9 +105,36 @@ def __call__(self, images: Union[str, List[str], "Image.Image", List["Image.Imag
         """
         return super().__call__(images, **kwargs)
 
-    def preprocess(self, image):
+    def preprocess(self, image, prompt=None):
         image = load_image(image)
-        model_inputs = self.image_processor(images=image, return_tensors=self.framework)
+
+        if prompt is not None and not isinstance(prompt, str):
+            raise ValueError(
+                f"Received an invalid text input, got - {type(prompt)} - but expected a single string. "
+                "Note also that one single text can be provided for conditional image to text generation."
+            )
+
+        elif prompt is not None and self.model.config.model_type == "git":
+            model_inputs = self.image_processor(images=image, return_tensors=self.framework)
+            input_ids = self.tokenizer(text=prompt, add_special_tokens=False).input_ids
+            input_ids = [self.tokenizer.cls_token_id] + input_ids
+            input_ids = torch.tensor(input_ids).unsqueeze(0)
+            model_inputs.update({"input_ids": input_ids})
+
+        elif prompt is not None and self.model.config.model_type == "pix2struct":
+            model_inputs = self.image_processor(images=image, header_text=prompt, return_tensors=self.framework)
+
+        elif prompt is not None and self.model.config.model_type != "vision-encoder-decoder":
+            # vision-encoder-decoder does not support conditional generation
+            model_inputs = self.image_processor(images=image, return_tensors=self.framework)
+            text_inputs = self.tokenizer(prompt, return_tensors=self.framework)
+            model_inputs.update(text_inputs)
+        else:
+            model_inputs = self.image_processor(images=image, return_tensors=self.framework)
+
+        if self.model.config.model_type == "git" and prompt is None:
+            model_inputs["input_ids"] = None
```

Comment:
`prompt` is not None is tested in every branch here. Can we do a bit `if prompt is not None` then the cascade of tests (which indicates all those models should probably not be in the same Auto class by the way, since they all have different APIs) and then the git part in the else branch? It would be more readable this way.